airbyte-integrations/connectors/source-shopify/source_shopify/source.py

```diff
@@ -48,10 +49,75 @@ class ShopifyStream(HttpStream, ABC):
 
     def __init__(self, shop: str, start_date: str, api_password: str, **kwargs):
         super().__init__(**kwargs)
+        self._schema = self.get_json_schema()
         self.start_date = start_date
         self.shop = shop
         self.api_password = api_password
 
+    @staticmethod
+    def _get_json_types(value_type) -> str:
+        json_types = {
+            str: ["string"],
+            int: ["integer", "number"],
+            float: ["number"],
+            dict: ["object"],
+            list: ["array"],
+            bool: ["boolean"],
+            type(None): ["null", ]
+        }
+        return json_types.get(value_type)
+
+    @staticmethod
+    def _convert(value: Any, convert_type: str):
+        if convert_type == "number":
+            value = Decimal(value)
+        return value
+
+    @staticmethod
+    def _find_schema_type(schema_types: List[str]) -> str:
+        not_null_types = schema_types.copy()
+        if "null" in not_null_types:
+            not_null_types.remove("null")
+        return not_null_types[0]
+
+    def _transform_array(self, array: List[Any], item_properties: Mapping[str, Any]):
+        item_types = item_properties.get("type", [])
+        if item_types:
+            schema_type = self._find_schema_type(item_types)
+            nested_properties = item_properties.get("properties", {})
+            for item in array:
+                if schema_type == "object":
+                    self._transform_object(item, nested_properties)
+
+    def _transform_object(self, transform_object: Mapping[str, Any], properties: Mapping[str, Any]):
```

Comment:
in general, it is hard to track what is going on, and why are we doing the transformation.
Nice clear comments could help